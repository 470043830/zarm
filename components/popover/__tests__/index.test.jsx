import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popover from '../index';

describe('Popover', () => {
  it('renders correctly', () => {
    const wrapper = render(<Popover message="hello"><span>foo</span></Popover>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('visible', () => {
    jest.useFakeTimers();
    const wrapper = mount(<Popover visible message="hello"><span>foo</span></Popover>);
    expect(toJson(wrapper)).toMatchSnapshot();
    jest.runAllTimers();
    wrapper.unmount();
  });
});
