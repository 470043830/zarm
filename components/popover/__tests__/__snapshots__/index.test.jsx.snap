// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover renders correctly 1`] = `
<span>
  foo
</span>
`;

exports[`Popover visible 1`] = `
<Popover
  message="hello"
  prefixCls="za-popover"
  visible={true}
>
  <span>
    foo
  </span>
</Popover>
`;
